tests/test_nn.py

```diff
@@ -51,14 +51,20 @@ def test_linear_float():
     print(s.build(target="vhls"))
 
 
+def np_softmax(x, axis=-1):
+    x_max = np.max(x, axis=axis, keepdims=True)
+    e_x = np.exp(x - x_max)
+    return e_x / np.sum(e_x, axis=axis, keepdims=True)
+
+
 def test_softmax():
     from allo.library.nn import softmax
 
     s = allo.customize(softmax, instantiate=[float32, 8])
     mod = s.build()
     inp = np.random.randn(8, 8).astype(np.float32)
```

Comment:
Can you also use a larger test case? I remember you used some large random tensors in your own test.

Reply:
Sure! I added `inp = 1000 * inp` to create larger tensor test cases.